src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -1178,5 +1178,195 @@ private class MyClassCustomConverterFactory : JsonConverterFactory
+
+            Assert.False(jpi.IsExtensionData);
+            jpi.IsExtensionData = true;
+            Assert.True(jpi.IsExtensionData);
+        }
+
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(ClassWithExtensionDataAttribute))]
+        [InlineData(typeof(List<int>))]
+        [InlineData(typeof(IDictionary<string, string>))]
+        [InlineData(typeof(Dictionary<string, int>))]
+        public static void EnablingExtensionData_UnsupportedPropertyType_ThrowsInvalidOperationException(Type type)
+        {
+            var jti = JsonTypeInfo.CreateJsonTypeInfo(typeof(ClassWithExtensionDataAttribute), new());
+            var jpi = jti.CreateJsonPropertyInfo(type, "ExtensionData");
+
+            Assert.False(jpi.IsExtensionData);
+            Assert.Throws<InvalidOperationException>(() => jpi.IsExtensionData = true);
+        }
+
+        public class ClassWithExtensionDataAttribute
+        {
+            public int Value1 { get; set; }
+            public int Value2 { get; set; }
+
+            [JsonExtensionData]
+            public Dictionary<string, object> ExtensionData { get; set; }
+        }
+
+        [Fact]
+        public static void DefaultJsonTypeInfoResolver_JsonPropertyInfo_ReturnsMemberInfoAsAttributeProvider()
+        {
+            var resolver = new DefaultJsonTypeInfoResolver();
+            JsonTypeInfo jti = resolver.GetTypeInfo(typeof(ClassWithFieldsAndProperties), new());
+
+            Assert.Equal(2, jti.Properties.Count);
+
+            JsonPropertyInfo fieldPropInfo = jti.Properties.First(prop => prop.Name == "Field");
+            FieldInfo fieldInfo = typeof(ClassWithFieldsAndProperties).GetField("Field");
+            Assert.NotNull(fieldInfo);
+            Assert.Same(fieldInfo, fieldPropInfo.AttributeProvider);
+
+            JsonPropertyInfo propertyPropInfo = jti.Properties.First(prop => prop.Name == "Property");
+            PropertyInfo propInfo = typeof(ClassWithFieldsAndProperties).GetProperty("Property");
+            Assert.NotNull(propInfo);
+            Assert.Same(propInfo, propertyPropInfo.AttributeProvider);
+        }
+
+        [Fact]
+        public static void JsonPropertyInfo_AttributeProvider_ReassigningValuesHasNoEffect()
+        {
+            var options = new JsonSerializerOptions
+            {
+                TypeInfoResolver = new DefaultJsonTypeInfoResolver()
+                {
+                    Modifiers =
+                    {
+                        jti =>
+                        {
+                            if (jti.Type == typeof(ClassWithFieldsAndProperties))
+                            {
+                                Assert.Equal(2, jti.Properties.Count);
+                                jti.Properties[0].AttributeProvider = jti.Properties[1].AttributeProvider;
+                                jti.Properties[1].AttributeProvider = null;
```

Comment:
you might want to add assert which checks that what you assigned is what you get

Reply:
Sure.